nectar/downloaders/threaded.py

```diff
@@ -215,10 +215,14 @@ def _fetch(self, request, session):
                             url=request.url)
                         )
                         _logger.warning(msg)
-                    response = session.get(
-                        request.url, headers=headers, timeout=(self.config.connect_timeout,
-                                                               self.config.read_timeout)
-                    )
+
+                    response = session.get(request.url, headers=headers,
+                                           timeout=(self.config.connect_timeout,
+                                                    self.config.read_timeout))
+
+                    report.headers = response.headers
```

Comment:
@ipanova Is there any chance that response.headers won't be available when this runs? For example say the webserver is really slow and doesn't send the headers quickly. This code will execute and could it just save a reference to response.headers, or maybe it's None? I think if you use Nectar to make a request to a pulp URL that has a `pdb` statement in it which prevents the response from being formed, then we should wait to see the `download_headers` event until after the response (including the headers) has been returned from Django.